src/Algebra/Graph/Relation/InternalDerived.hs

```diff
@@ -45,48 +48,167 @@ instance Ord a => Eq (ReflexiveRelation a) where
+            filtered (Relation d r) = Relation d (Set.filter (uncurry (<=)) r)
+
+instance Ord a => Ord (SymmetricRelation a) where
+    compare x y = mconcat
+        [ compare (Set.size . domain . fromSymmetric $   x) (Set.size . domain . fromSymmetric $   y)
+        , compare (           domain . fromSymmetric $   x) (           domain . fromSymmetric $   y)
+        , compare (Set.size . Set.filter (uncurry (<=)) . relation . fromSymmetric $ x) 
+                  (Set.size . Set.filter (uncurry (<=)) . relation . fromSymmetric $ y)
+        , compare (           relation . fromSymmetric $ x) (           relation . fromSymmetric $ y) ]
+
+-- | Construct the /empty graph/.
+-- Complexity: /O(1)/ time and memory.
+--
+-- @
+-- 'Algebra.Graph.Relation.Symmetric.isEmpty'     empty == True
+-- 'Algebra.Graph.Relation.Symmetric.hasVertex' x empty == False
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' empty == 0
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   empty == 0
+-- @
+empty :: SymmetricRelation a
+empty = SymmetricRelation $ Relation Set.empty Set.empty
+
+-- | Construct the graph comprising /a single isolated vertex/.
+-- Complexity: /O(1)/ time and memory.
+--
+-- @
+-- 'Algebra.Graph.Relation.Symmetric.isEmpty'     (vertex x) == False
+-- 'Algebra.Graph.Relation.Symmetric.hasVertex' x (vertex x) == True
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (vertex x) == 1
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (vertex x) == 0
+-- @
+vertex :: a -> SymmetricRelation a
+vertex x = SymmetricRelation $ Relation (Set.singleton x) Set.empty
+
+-- | /Overlay/ two graphs. This is a commutative, associative and idempotent
+-- operation with the identity 'empty'.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- 'Algebra.Graph.Relation.Symmetric.isEmpty'     (overlay x y) == 'Algebra.Graph.Relation.Symmetric.isEmpty'   x   && 'Algebra.Graph.Relation.Symmetric.isEmpty'   y
+-- 'Algebra.Graph.Relation.Symmetric.hasVertex' z (overlay x y) == 'Algebra.Graph.Relation.Symmetric.hasVertex' z x || 'Algebra.Graph.Relation.Symmetric.hasVertex' z y
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (overlay x y) >= 'Algebra.Graph.Relation.Symmetric.vertexCount' x
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (overlay x y) <= 'Algebra.Graph.Relation.Symmetric.vertexCount' x + 'Algebra.Graph.Relation.Symmetric.vertexCount' y
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (overlay x y) >= 'Algebra.Graph.Relation.Symmetric.edgeCount' x
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (overlay x y) <= 'Algebra.Graph.Relation.Symmetric.edgeCount' x   + 'Algebra.Graph.Relation.Symmetric.edgeCount' y
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (overlay 1 2) == 2
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (overlay 1 2) == 0
+-- @
+overlay :: Ord a => SymmetricRelation a -> SymmetricRelation a -> SymmetricRelation a
+overlay x y = SymmetricRelation $ Relation (domain (fromSymmetric x) 
```

Comment:
Better pattern-match on `SymmetricRelation x` to avoid doing `fromSymmetric` multiple times. Same in `connect`.